src/ORM/Behavior/TreeBehavior.php

```diff
@@ -76,6 +78,15 @@ class TreeBehavior extends Behavior
     ];
 
     /**
+     * {@inheritDoc}
+     */
+    public function initialize(array $config)
+    {
+        $this->_config['leftField'] = new IdentifierExpression($this->_config['left']);
+        $this->_config['rightField'] = new IdentifierExpression($this->_config['right']);
```

Comment:
As @skie said, it is almost impossible to override config() for this purpose. It is either this, or creating the expression object on each method